_studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp

```diff
@@ -906,10 +906,32 @@ mfxStatus VideoDECODEAV1::SubmitFrame(mfxBitstream* bs, mfxFrameSurface1* surfac
                  m_video_par.AsyncDepth = static_cast<mfxU16>(vp.async_depth);
             }
 
-            switch (umcRes)
+            if (umcRes == UMC::UMC_WRN_REPOSITION_INPROGRESS)
```

Comment:
Ok.... So, we check umcRes here for UMC::UMC_WRN_REPOSITION_INPROGRESS while we entered condition on the same status right above... and we overrode the status... Question: do we expect that m_decoder->GetInfo will return same status? why? Honestly, I am not sure which status all the code below actually handles... which is not good...

Reply:
Seems like this should never happen, according to GetInfo implementation for AV1.
https://github.com/Intel-Media-SDK/MediaSDK/blob/52c71f460abea29fce1dea69df2b67c8d869a80e/_studio/shared/umc/codec/av1_dec/src/umc_av1_decoder.cpp#L141-L151
Because it returns different statuses.